### PR TITLE
[RUBY-2704] Fix scoping issues in `Payment` model and update tests

### PR DESCRIPTION
**Pull request by jjromeo**
- Extend `Payment` class from `ApplicationRecord` for ActiveRecord functionality.
- Update `refunds_and_reversals` and `refundable` scopes to include `.success` for filtering successful payments.
- Add comprehensive tests for all payment scopes, ensuring correct filtering and ordering of payments.
- Refactor test setup for payment types and status to improve clarity and maintainability.

### CODE REVIEW

File: spec/models/waste_exemptions_engine/payment_spec.rb [hunk 1]
```diff
@@ -1,28 +1,29 @@
 # frozen_string_literal: true
 
 require "rails_helper"
+require WasteExemptionsEngine::Engine.root.join("app", "models", "waste_exemptions_engine", "payment")
```
Comment (PaulDoyle-EA) [inline spec/models/waste_exemptions_engine/payment_spec.rb]: Why is this necessary?

File: spec/models/waste_exemptions_engine/payment_spec.rb [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
-    let(:account) { create(:account, registration:) }
-
     describe "#maximum_refund_amount" do
+      let(:payment_amount) { 100 }
+      let(:payment) { create(:payment, :success, payment_amount:) }
+
       context "when payment type is not refundable" do
-        let(:reversal_payment) { create(:payment, payment_type: Payment::PAYMENT_TYPE_REVERSAL, account:) }
+        before do
+          allow(payment).to receive(:payment_type).and_return(Payment::PAYMENT_TYPE_REVERSAL)
```
Comment (PaulDoyle-EA) [inline spec/models/waste_exemptions_engine/payment_spec.rb]: Can this be done in the `let` statement for the payment? Or as a trait in the factory?

File: spec/models/waste_exemptions_engine/payment_spec.rb (hunk 1 above)
Comment (jjromeo) [reply spec/models/waste_exemptions_engine/payment_spec.rb]: It's not, added while trying to debug the sonarcloud issue! Removed.

**Review by PaulDoyle-EA — APPROVED** [review]
